schemas/telemetry/untrusted-modules/untrusted-modules.4.parquetmr.txt

```diff
@@ -0,0 +1,313 @@
+        repeated group list {
+          required binary element (UTF8);
+        }
+      }
+      optional group antispyware (LIST) {
+        repeated group list {
+          required binary element (UTF8);
+        }
+      }
+      optional group antivirus (LIST) {
+        repeated group list {
+          required binary element (UTF8);
+        }
+      }
+    }
+    optional group hdd {
+      optional group profile {
+        optional binary model (UTF8);
+        optional binary revision (UTF8);
+      }
+      optional group binary {
+        optional binary model (UTF8);
+        optional binary revision (UTF8);
+      }
+      optional group system {
+        optional binary model (UTF8);
+        optional binary revision (UTF8);
+      }
+    }
+    optional group cpu {
+      optional int64 cores;
+      optional int64 count;
+      optional int64 family (UINT_64);
+      optional int64 l2cacheKB (UINT_64);
+      optional int64 l3cacheKB (UINT_64);
+      optional int64 model (UINT_64);
+      optional int64 speedMHz (UINT_64);
+      optional int64 stepping (UINT_64);
+      optional binary vendor (UTF8);
+      optional group extensions (LIST) {
+        repeated group list {
+          required binary element (UTF8);
+        }
+      }
+    }
+    required group build {
+      required binary applicationId (UTF8);
+      required binary applicationName (UTF8);
+      required binary architecture (UTF8);
+      optional binary architecturesInBinary (UTF8);
+      required binary buildId (UTF8);
+      optional binary hotfixVersion (UTF8);
+      required binary version (UTF8);
+      required binary vendor (UTF8);
+      optional binary displayVersion (UTF8);
+      required binary platformVersion (UTF8);
+      optional boolean updaterAvailable;
+      required binary xpcomAbi (UTF8);
+    }
+    required group partner {
+      optional binary distributionId (UTF8);
+      optional binary distributionVersion (UTF8);
+      optional binary partnerId (UTF8);
+      optional binary distributor (UTF8);
+      optional binary distributorChannel (UTF8);
+      optional group partnerNames (LIST) {
+        repeated group list {
+          optional binary element (UTF8);
+        }
+      }
+    }
+    required group addons {
+      optional binary persona (UTF8);
+      required group theme {
+        required binary id (UTF8);
+        optional boolean blocklisted;
+        optional binary description (UTF8);
+        optional binary name (UTF8);
+        optional boolean userDisabled;
+        optional boolean appDisabled;
+        optional binary version (UTF8);
+        optional int64 scope;
+        optional boolean hasBinaryComponents;
+        optional int64 installDay (UINT_64);
+        optional int64 updateDay (UINT_64);
+      }
+      required group activeAddons (MAP) {
+        repeated group key_value {
+          required binary key (UTF8);
+          required group value {
+            required boolean blocklisted;
```

Comment:
It doens't look like any of these are required by the JSON schema

Reply:
Thanks for catching this; I corrected the required / optional for these and a few other properties based off the JSON schema.